generic_tracers/generic_COBALT.F90

```diff
@@ -1,53 +1,52 @@
+!  in these processes, while being mindful of computational efficiency and
+!  the desire to rely on dynamics which are well supported by empirical studies.
+!  In some cases, this means forgoing less understood interactions of uncertain
+!  quantitative importance to establish a robustly supported baseline simulation
+!  until constraints improve.  The COBALT code releases reflect this approach:
 !
-!       Stock, CA, Dunne, JP, John, JG. 2014. Global-scale carbon and energy
-!         flows through the marine planktonic food web: An analysis with a'
-!         coupled physical-biological model.  Progress in Oceanography 120, 1-18.
+!  COBALTv1 (Stock et al., 2014) augmented the plankton foodweb dynamics of
+!  GFDL's prior biogeochemical model (TOPAZ, Dunne et al., 2013) by adding 3
+!  zooplankton groups, free living bacteria, and expanding the resolution of
+!  dissolved organic material to include labile, semi-labile and semi-refractory
+!  forms.  This produced quantitatively credible estimates of energy flows through
+!  plankton foodwebs.
 !
-!       Version 2.0 has a number of refinements:
+!  COBALTv2 served as the ocean biogeochemical component of GFDL's CMIP6 earth
+!  system model (Stock et al., 2020).  The structure of COBALTv2 was largely
+!  analogous to COBALTv1, but notable adjustments included: 1) the addition of
+!  temperature and O2-dependent particle remineralization following Laufkotter
+!  et al. (2017); 2) retuning of ammonia dynamics and the addition air-sea ammonia
+!  exchanges (Paulot et al., 2020); 3) adjustments to iron sources and
+!  scavenging; and 4) movement of the defaul carbon chemistry to MOCSY.
 !
-!           1) Ammonia uptake parameters are now based on the "high-affinity"
-!              settings from Paulot et al., 2015; GBC; 29(8)
-!           2) Phytoplankton aggregation is initiated only when growth rates
-!              fall below 1/4 maximum values
-!           3) The default parameterization has elevated N:P ratios for both
-!              diazotrophs and small phytoplankton
-!           4) Remineralization of sinking detritus is now based on the temperature
-!              and oxygen dependences described in Laufkotter et al., 2017; O2
-!              dependence of other aerobic processes have also been adjusted
-!              for consistency.
-!           5) The default carbon chemistry routine is now MOCSY
-!           6) The iron scavenging has been re-tuned to new atmospheric (Ginoux-AM4),
-!              sediment, river and hydrothermal vent (Tagliabue) sources.
-!
-!       parameterizations used herein, as well as definitions for all variables
-!       and parameters.  The 33 model state variables are:
+!  COBALTv3 (Stock et al., submitted) was developed to enhance the robustness of COBALT
+!  across open ocean and coastal regions.  It now divides the phytoplankton community
+!  into 3 rather than 2 size classes to better resolve phytoplankton commmunities from
```

Comment:
commmunities -> communities.

Should we also consider rephrasing this sentence to reflect that COBALTv3 includes small, medium, and large phytoplankton, as well as a diazotroph?

Reply:
Thanks for checking this Yi-Cheng.  How about we add the sentence: "Thus, COBALTv3 now includes small, medium, and large phytoplankton, as well as a diazotroph." after the Van Oostende reference.  The existing sentence clarifies the rationale behind the change, while the second ensures that the implications for the state variables are clear.

What is the quickest way/best process to modify a pull request?

Reply:
@charliestock, sounds good. In your case, you can simply use the GitHub UI to edit the file and save the change (see the screenshot below):

![edit](https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/ocean_BGC/assets/143743249/6c608902-12c8-4e32-a0df-0679a714a53c)

Reply:
Thanks for the pointer.  I think I managed to implement the suggested changes successfully, and I captured one additional typo.  Can you see the changes?